src/kaleidoscope/key_events.cpp

```diff
@@ -81,8 +81,13 @@ void handleKeyswitchEvent(Key mappedKey, byte row, byte col, uint8_t keyState) {
 
     /* If a key had an on event, we update the live composite keymap. See
      * layers.h for an explanation about the different caches we have. */
-    if (keyToggledOn(keyState))
-      Layer.updateLiveCompositeKeymap(row, col);
+    if (keyToggledOn(keyState)) {
+      if (mappedKey.raw == Key_NoKey.raw) {
```

Comment:
Possible workaround for whatever it is that breaks OneShot – change this line to:
```
  if (mappedKey.raw == Key_NoKey.raw || keyState & INJECTED) {
```